src/userobjects/Map2LDelem.C

```diff
@@ -0,0 +1,131 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Map2LDelem.h"
+#include "MooseMesh.h"
+registerMooseObject("DeerApp", Map2LDelem);
+
+defineLegacyParams(Map2LDelem);
+
+InputParameters Map2LDelem::validParams() {
+  InputParameters params = InterfaceUserObject::validParams();
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "ld_block_names", "The name of the lower dimensional element blocks");
+  params.addClassDescription(
+      "A userobject creating the map between LD elements and solid element "
+      "number and side");
+  return params;
+}
+
+Map2LDelem::Map2LDelem(const InputParameters &parameters)
+    : InterfaceUserObject(parameters),
+      ld_block_ids(_mesh.getSubdomainIDs(
+          getParam<std::vector<SubdomainName>>("ld_block_names"))) {}
+
+Map2LDelem::~Map2LDelem() {}
+
+void Map2LDelem::initialSetup() {
```

Comment:
This method is way too big.  Split up into logical parts (use private member functions if that makes sense).